templates/repo/view_file.tmpl

```diff
@@ -30,41 +30,65 @@
+						<a class="ui mini basic button" href="{{.RepoLink}}/src/commit/{{PathEscape .CommitID}}/{{PathEscapeSegments .TreePath}}">{{.i18n.Tr "repo.file_permalink"}}</a>
+					{{end}}
+					{{if .IsRepresentableAsText}}
+						<a class="ui mini basic button" href="{{.RepoLink}}/blame/{{.BranchNameSubURL}}/{{PathEscapeSegments .TreePath}}">{{.i18n.Tr "repo.blame"}}</a>
+					{{end}}
+					<a class="ui mini basic button" href="{{.RepoLink}}/commits/{{.BranchNameSubURL}}/{{PathEscapeSegments .TreePath}}">{{.i18n.Tr "repo.file_history"}}</a>
+					{{if .EscapeStatus.Escaped}}
+						<a class="ui mini basic button unescape-button" style="display: none;">{{.i18n.Tr "repo.unescape_control_characters"}}</a>
+						<a class="ui mini basic button escape-button">{{.i18n.Tr "repo.escape_control_characters"}}</a>
+					{{end}}
+				</div>
+				<a download href="{{$.RawFileLink}}"><span class="btn-octicon tooltip" data-content="{{.i18n.Tr "repo.download_file"}}" data-position="bottom center">{{svg "octicon-download"}}</span></a>
+				{{if .Repository.CanEnableEditor}}
+					{{if .CanEditFile}}
+						<a href="{{.RepoLink}}/_edit/{{PathEscapeSegments .BranchName}}/{{PathEscapeSegments .TreePath}}"><span class="btn-octicon tooltip" data-content="{{.EditFileTooltip}}" data-position="bottom center">{{svg "octicon-pencil"}}</span></a>
+					{{else}}
+						<span class="btn-octicon tooltip disabled" data-content="{{.EditFileTooltip}}" data-position="bottom center">{{svg "octicon-pencil"}}</span>
+					{{end}}
+					{{if .CanDeleteFile}}
+						<a href="{{.RepoLink}}/_delete/{{PathEscapeSegments .BranchName}}/{{PathEscapeSegments .TreePath}}"><span class="btn-octicon btn-octicon-danger tooltip"  data-content="{{.DeleteFileTooltip}}" data-position="bottom center">{{svg "octicon-trash"}}</span></a>
+					{{else}}
+						<span class="btn-octicon tooltip disabled" data-content="{{.DeleteFileTooltip}}" data-position="bottom center">{{svg "octicon-trash"}}</span>
+					{{end}}
 				{{end}}
+			{{else if .EscapeStatus.Escaped}}
+				<a class="ui mini basic button unescape-button mr-2" style="display: none;">{{.i18n.Tr "repo.unescape_control_characters"}}</a>
+				<a class="ui mini basic button escape-button mr-2">{{.i18n.Tr "repo.escape_control_characters"}}</a>
 			{{end}}
 		</div>
-		{{end}}
 	</h4>
 	<div class="ui attached table unstackable segment">
+		{{if .EscapeStatus.BadBIDI}}
+			<div class="ui error message file-message">
+				<span class="close icon">{{svg "octicon-x" 16 "close inside"}}</span>
+				<div class="header">
+					{{.i18n.Tr "repo.bidi_bad_header"}}
+				</div>
+				<p>{{.i18n.Tr "repo.bidi_bad_description" | Str2html}}</p>
+			</div>
+		{{else if .EscapeStatus.Escaped}}
+			<div class="ui warning message file-message">
+				<span class="close icon">{{svg "octicon-x" 16 "close inside"}}</span>
+				<div class="header">
+					{{.i18n.Tr "repo.unicode_header"}}
+				</div>
+				<p>{{.i18n.Tr "repo.unicode_description" | Str2html}}</p>
```

Comment:
Can we put the code block into a template? Then we can include the template for all cases, and no need to copy&paste the code block.